src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_padding.cpp

```diff
@@ -14,6 +15,49 @@ using namespace cldnn;
 using namespace ov::intel_gpu;
 
 void prepare_padding::run(program& p) {
+    if (p.get_config().get_use_onednn()) {
+        for (const auto& node : p.get_processing_order()) {
+            if (!node->is_type<fully_connected>())
+                continue;
+
+            if (node->get_preferred_impl_type() != impl_types::onednn)
+                continue;
+
+            auto& input_node = node->get_dependency(1);
+            if (input_node.is_type<data>()
+                && input_node.is_constant()
+                && !input_node.is_dynamic()
```

Comment:
dynamism check is redundant because it is already data or constant

Reply:
Done.